pom.xml

```diff
@@ -1972,14 +1972,12 @@
             </execution>
             <execution>
               <id>scala-compile-first</id>
-              <phase>process-resources</phase>
               <goals>
                 <goal>compile</goal>
               </goals>
             </execution>
             <execution>
               <id>scala-test-compile-first</id>
-              <phase>process-test-resources</phase>
```

Comment:
Changing phase to one before `compile` is very old trick, used when `scala-maven-plugin` compiles only Scala classes and this compilation must be performed before Java classes compilation (by `maven-compiler-plugin`) because Java classes depend on Scala classes.
In incremental compilation mode SBT compiler is used internally and it compiles everything. Because `maven-compiler-plugin` is disabled (compilations are skipped), there is no need to change default phase.